packages/nextjs/src/server/data/getAuthDataFromRequest.ts

```diff
@@ -53,5 +58,9 @@ export function getAuthDataFromRequest(
     authObject = signedInAuthObject(options, jwt.raw.text, jwt.payload);
   }
 
+  if (treatPendingAsSignedOut && authObject.sessionStatus === 'pending') {
+    authObject = signedOutAuthObject(options, authObject.sessionStatus);
+  }
```

Comment:
Same comment as the Astro PR, I would like to see this come from `@clerk/backend`:

```ts
getAuthObjectFromJwt(jwt)
```

Reply:
Agreed! I'll raise a follow-up PR for it.